test/lifecycle_test/component2.dart

```diff
@@ -133,6 +175,19 @@ class _LifecycleTestWithContext extends _LifecycleTest {
   get contextType => LifecycleTestContext;
 }
 
+ReactDartComponentFactoryProxy2 ErrorComponent = react.registerComponent(() => new _ErrorComponent());
```

Comment:
Should this be `registerComponent2`?

Reply:
Currently `setClientConfiguration` is set to use `registerComponent` rather than `registerComponent2`. As a result, using `registerComponent2` throws the `setClientConfiguration must be called before registerComponent.` exception. That makes me think it would take a notable workaround to get `registerComponent2` to work, whereas `registerComponent` already has a short circuit that redirects to `registerComponent2` if the component is of the type `Component2`. Did I miss something and it would actually be straightforward or is this ok for now?